# 01 - Python para Data science -1

## Alumno: Pablo Matías Monsalvo G8-ONE




###  02 Para saber más: ¿Qué es Python?

Python es un lenguaje de programación de alto nivel, interpretado y orientado a objetos. Fue creado a finales de 1989 por el programador holandés Guido van Rossum. El nombre "Python" se inspiró en la serie de televisión británica Monty Python's Flying Circus y tiene como símbolo la imagen que se muestra a continuación:

![python.webp](/home/pol/Escritorio/ipynb-alura/assets/python.webp)

Python se utiliza en diversas áreas, como el desarrollo de aplicaciones web, la ciencia de datos, el aprendizaje automático, la automatización de tareas, el análisis de datos y muchas otras. Es un lenguaje de programación versátil que se puede utilizar para crear prácticamente cualquier tipo de programa.

Debido a su sintaxis simple y fácil de leer, se convierte en una buena elección para principiantes en la programación. Además, Python es uno de los lenguajes más utilizados en el mundo, con una gran comunidad activa y muchas bibliotecas y frameworks disponibles, lo que facilita la realización de diversas tareas.

Es compatible con varios sistemas operativos, como Windows, Linux y macOS. Como es un lenguaje popular y en constante evolución, Python se utiliza en muchas empresas y organizaciones en todo el mundo. También se utiliza ampliamente en aplicaciones científicas debido a su capacidad para manejar grandes conjuntos de datos y operaciones matemáticas complejas.

La información y la documentación sobre Python se pueden encontrar en el sitio web oficial python.org.

###  03 Preparando el ambiente

Vamos a acceder al notebook que será utilizado durante el curso directamente desde github. Una vez accedes al github has clic sobre el siguiente botón para que lo puedas abrir directamente en tu cuenta de Colab:

1.jpg

Como lo acabamos de mencionar, nuestro ambiente de trabajo será Google Colaboratory. Para lograr acceder a este, es necesario tener una cuenta en Gmail, dado que el notebook quedará almacenado en Google Drive.

###  04 Google Colaboratory

¡Hola! En esta clase, se introduce Python como un lenguaje de programación multiparadigma, creado por Guido Van Roosum en los 90. Se explica la diferencia entre lenguajes interpretados (como Python) y compilados, utilizando una analogía de una receta en japonés.

Además, se destaca la idoneidad de Python para Data Science debido a su sintaxis simple y la gran comunidad que ha desarrollado herramientas como IPython. Se presenta Google Colab como una plataforma en línea para trabajar con notebooks de IPython, facilitando la experimentación y el aprendizaje de Python sin necesidad de instalación local.

Finalmente, se explora la interfaz de Google Colab, incluyendo la creación de celdas de código y texto, el uso de Markdown para formatear texto, y atajos de teclado para una edición más eficiente.

###  05 Para saber más: ¿Qué es Google Colaboratory?

Google Colab es una herramienta que permite escribir y ejecutar código en Python a través de un navegador, sin necesidad de instalar ningún software adicional. Funciona mediante el uso de una máquina virtual, que es esencialmente un sistema virtual que simula una computadora.

Cuando creamos, cargamos o subimos un nuevo documento en Colab, estamos creando una máquina virtual que se ejecuta en uno de los servidores de Google. Esta máquina virtual tiene acceso a una serie de recursos, como CPU, memoria y almacenamiento, que se utilizan para ejecutar el código.

Es importante destacar que esta máquina virtual es temporal, lo que significa que se crea cuando abrimos Colab y se cierra cuando lo cerramos o cuando expira el tiempo de ejecución de la máquina. Esto significa que cualquier dato que carguemos en la máquina virtual se perderá cuando se cierre.

Puedes darte cuenta de que tu máquina virtual se ha cerrado cuando el indicador de conexión en la esquina superior derecha está en espera para conectarse, como se muestra en la imagen a continuación:

![google-colab.jpg](/home/pol/Escritorio/ipynb-alura/assets/google-colab.jpg)

Para volver a conectar la máquina, simplemente haz clic en el botón "Conectar" y espera a que se establezca una nueva conexión. También es necesario ejecutar todas las celdas de creación y manipulación de variables que se crearon y ejecutaron antes de que la máquina virtual se cerrara, para que puedan ser utilizadas.

###  06 Hola mundo

¡Hola! En esta clase, se exploran los primeros pasos con Python en el entorno de Colab.

Aquí tienes un resumen de los puntos clave:

Ejecución de celdas: Se explica cómo ejecutar celdas de código con Shift + Enter (ejecuta y añade una nueva celda) y Ctrl + Enter (ejecuta sin añadir una nueva celda).
Hola Mundo: Se introduce el comando print("Hola, Mundo") como una convención para empezar a programar en cualquier lenguaje, destacando la simplicidad de Python en comparación con otros lenguajes como Java.
Función print(): Se explora la función print() como una función incorporada de Python que permite mostrar texto u otros valores en la salida. Se menciona la documentación de Python y las Built-in Functions para facilitar el aprendizaje y la utilización del lenguaje.
Impresión de texto y valores: Se muestra cómo imprimir texto utilizando paréntesis simples o dobles y cómo imprimir valores numéricos directamente.
Ejecución línea por línea: Se explica cómo Python interpreta el código línea por línea, lo que permite ver la salida de cada comando de forma inmediata.

In [31]:
print('hola mundo soy pol Monsalvo')

hola mundo soy pol Monsalvo


###  07 ¿Qué hace la función print()?

Aprendimos sobre una de las funciones de Python: La función print(). Selecciona la alternativa que describe lo que esta función hace y muestra un ejemplo usando su sintaxis.

La función print se utiliza para mostrar información en la pantalla, es decir, imprime o muestra el contenido pasado como argumento. Un ejemplo de uso de esta función es el siguiente código: print('¡Hola, mundo!')


La función print es una de las más útiles en el desarrollo de programas en Python, ya que permite ver lo que está sucediendo en su programa y ayuda a encontrar errores. Su sintaxis es simple: escribimos print seguido de paréntesis () y dentro de ellos ponemos lo que deseamos imprimir, como se muestra en la alternativa.

###  08 Desafío: ¡Es hora de practicar!

Vamos a practicar el uso de la función print con algunas actividades. Para ello, resuelve los problemas propuestos en código:

* 1 - Imprime la frase "Escuela de Datos de Alura".

* 2 - Imprime tu nombre y tu apellido siguiendo la estructura a continuación:

Nombre: [tu nombre]
Apellido: [tu apellido]

* 3 - Imprime tu primer nombre letra por letra. Por ejemplo, si mi nombre es Álvaro, la salida sería:

Á
L
V
A
R
O

* 4 - Imprime tu fecha de nacimiento en formato día mes año. Recuerda que los valores de día y año no deben estar entre comillas. Suponiendo una fecha de cumpleaños el 28 de febrero de 2003, el formato debe ser como el siguiente:

28 febrero 2003

* 5 - Imprime, en una sola instrucción print, el año actual en el que estás realizando este curso. El valor del año debe ser un dato numérico, y la salida de print debe ser la siguiente:

Año actual: [año]

Si necesitas ayuda, las soluciones a las actividades están disponibles en la sección "Opinión del instructor".

### Opinión del instructor

Pueden existir varias formas de resolver los problemas propuestos, y algunas de ellas se presentan a continuación.

Para imprimir la frase "Escuela de Datos de Alura", usamos la función print y colocamos el texto entre comillas simples dentro de los paréntesis:

In [32]:
print('Escuela de Datos de Alura')

Escuela de Datos de Alura


Cuando colocamos dos impresiones (print) en secuencia (una debajo de la otra), podemos obtener un salto de línea. Por lo tanto, el resultado esperado se puede obtener de la siguiente manera:

print('Nombre: <tu nombre>')
print('Apellido: <tu apellido>')
Siguiendo la misma lógica del ejercicio anterior, podemos separar líneas utilizando varias impresiones (print):

In [33]:
print('Nombre: Pablo')
print('Apellido: Monsalvo')


Nombre: Pablo
Apellido: Monsalvo


In [34]:
print('P')
print('a')
print('b')
print('l')
print('o')


P
a
b
l
o


Podemos combinar datos numéricos y textuales separándolos por comas. Así, podemos imprimir el día 28, seguido de una coma, luego el mes entre comillas 'febrero', otra coma y, finalmente, el año en formato numérico 2003:

In [35]:
print(7, 'junio', 2025)

7 junio 2025


Con el mismo razonamiento que en la pregunta anterior, es posible imprimir datos numéricos y textuales separándolos por comas. De esta manera, podemos imprimir la frase 'Año actual:' y el año actual dentro de la función print, separándolos por una coma. Suponiendo que estemos tomando este curso en el año 2023, la solución sería la siguiente:

In [36]:
print('Año actual: ', 2025)

Año actual:  2025


Es importante practicar para consolidar el contenido, desarrollar habilidades de codificación, identificar puntos que aún no se comprenden, prepararse para desafíos futuros y desarrollar el pensamiento lógico y la resolución de problemas.

Por lo tanto, es importante realizar muchos ejercicios cuando se está aprendiendo un nuevo lenguaje de programación. Así que busca practicar a través de los ejercicios propuestos y observa cómo te pueden ayudar a mejorar tus habilidades de codificación.



###  09 Lo que aprendimos


Lo que aprendimos en esta aula:

* Utilizar Google Colab para proyectos en Python.
* Aplicar la función print para imprimir resultados en la pantalla.
* Comprender qué es Python y conocer su historia.

-------------

## 02 - Manipulando datos en Python

### 02 - Variables

¡Hola! ¡Claro que sí!

En esta lección, se exploran los siguientes temas:

Entorno de Ejecución en Colab: Se explica cómo Colab asigna recursos de una máquina virtual para ejecutar Python y la importancia de mantener activo el entorno de ejecución.
Variables en Python: Se introduce el concepto de variables como nombres asociados a valores en la memoria. Se explica cómo Python asigna dinámicamente valores a variables y cómo la función id() revela la dirección de un objeto en la memoria.
Asignación de Variables: Se demuestra cómo asignar valores a variables utilizando el signo igual (=) y cómo esto crea una referencia al espacio de memoria donde se almacena el valor.
Tipos de Nombres de Variables: Se mencionan algunas restricciones para nombrar variables, como no comenzar con números, no usar espacios y evitar palabras reservadas de Python.
¿Te gustaría que profundicemos en alguno de estos temas o tienes alguna pregunta específica?

In [37]:
print('Mi nombre es Pablo y tengo 44 años y me encanta la ciencia de datos.')


Mi nombre es Pablo y tengo 44 años y me encanta la ciencia de datos.


In [38]:
85

85

In [39]:
id(85)

140101310008176

###  03 Para saber más: comentarios

A veces, cuando estamos programando, necesitamos describir alguna etapa, función o estructura dentro del propio código. Esta descripción debe darse como una anotación y, por lo tanto, no se considera código que se interprete en el entorno. Para estas situaciones, existen los comentarios.

Los comentarios son líneas de texto que se pueden colocar dentro del código y que son ignoradas por el intérprete. Por eso, sirven para explicar el funcionamiento del código a otras personas o incluso a la persona que programa. Son importantes para documentar el código, describiendo lo que hace y para qué sirve cada función, estructura o método. Esto puede hacer que el código sea más legible y fácil de entender.

Podemos construir dos tipos de comentarios en Python, de una y varias líneas.

Comentarios de una línea:

Estos comentarios se hacen agregando un símbolo de almohadilla/numeral (#) al principio de una línea de código. Todo lo que venga después del símbolo # en una línea se considerará un comentario, como se puede ver en el código a continuación:

In [40]:
# Este es un comentario de una línea
print(10) # Podemos colocar otro comentario en una línea después de un código

10


Puedes ejecutar el código anterior en tu cuaderno y verás que todo el texto después del símbolo # se ignora durante la ejecución.

### Comentarios de varias líneas:

Los comentarios de varias líneas se hacen usando un conjunto de comillas triples: ''' o """. Todo lo que esté entre estas comillas triples se considerará un comentario, incluso si abarca varias líneas. Por ejemplo:

In [41]:
'''
Este es un comentario
de varias líneas.
'''

'\nEste es un comentario\nde varias líneas.\n'

Mientras el texto esté dentro de las comillas, se ignorará durante la ejecución, ya sea en una línea de código o en cualquier otro texto. A pesar de estas formas de comentar el código, cuando trabajamos con un cuaderno de Python, tenemos la opción de crear celdas de texto en formato markdown.

### Markdown:

Markdown es una forma de formato de texto que permite escribir de manera simple y fácil de leer. Permite agregar formato al texto sin necesidad de escribir código HTML u otro lenguaje de formato más complejo.

Con markdown, podemos formatear el texto para agregar negritas, cursivas, listas, encabezados, etc. Además, es una excelente opción para documentar el código describiendo lo que hemos hecho, analizando los resultados y proporcionando pasos futuros. En Ciencia de Datos, se hace un amplio uso de textos en markdown, ya que facilita en gran medida la documentación en cuadernos.

Para crear una celda de texto en Google Colab, puedes hacer clic en el botón + Texto debajo de la barra superior de botones de Colab, como se muestra en la imagen a continuación:

![google-colab-2.jpg](/home/pol/Escritorio/ipynb-alura/assets/google-colab-2.jpg)

Al crear la celda de texto, puedes agregar el contenido que desees. El cuaderno de este curso está lleno de celdas de texto que explican varios aspectos de la programación en Python, todas formateadas con markdown. A continuación, te damos algunos consejos para construir texto con markdown:

### Encabezados:

Para construir encabezados, puedes usar el símbolo # antes del texto. Cuantos más símbolos # uses, menor será el nivel del encabezado. Por ejemplo:



In [42]:
# Encabezado de nivel 1
## Encabezado de nivel 2
### Encabezado de nivel 3

# Encabezado de nivel 1
## Encabezado de nivel 2
### Encabezado de nivel 3

### Negritas e itálicas:

Para formatear el texto en negritas, puedes usar ** antes y después del texto, como en el ejemplo:

 ** Texto en negrita ** 

 **Texto en negrita**



Para formatear el texto en cursiva, utiliza * antes y después del texto. Por ejemplo:

*Texto en cursiva *

*Texto en cursiva*

También puedes combinar ambas formas de escritura agrupando las dos estructuras con ** y *. Por ejemplo:

*** Texto en negritas y cursiva ***

***Texto en negrita y cursiva***

### Listas:

Para crear listas no numeradas (con viñetas), coloca * y luego un espacio * antes del elemento, como en el ejemplo:

* Elemento 1
* Elemento 2
* Elemento 3

Para crear listas numeradas, puedes usar un número seguido de un punto y un espacio 1. antes del elemento, como en el ejemplo:

1. Elemento 1
2. Elemento 2
3. Elemento 3

### Enlaces:

Para crear enlaces, a continuación, tienes un ejemplo con el enlace al sitio web de Alura:

[Página de Alura](https://www.aluracursos.com/)

Estos son ejemplos más comunes de uso de markdown, pero existen muchas otras formas y recursos disponibles para que domines el lenguaje de marcación.

##  04 Tipos de variables

¡Hola! En esta clase, aprendimos sobre los diferentes tipos de variables en Python:

Enteros (int): Números enteros como 5 o 11.
Flotantes (float): Números decimales como 9.5 o 9.75.
Cadenas (string): Textos como "Penelope Camacho".
Booleanos (bool): Valores de verdad como True o False.
También aprendimos a declarar variables en Python y a usar la función type() para verificar el tipo de una variable.

Finalmente, vimos cómo usar estos tipos de variables para guardar información de un estudiante, como su nombre, edad, promedio y situación académica.

¿Te gustaría que repasemos algún punto en particular o que veamos algunos ejemplos adicionales?


##  05 Variables numéricas

¡Hola! En esta clase, aprendimos a trabajar con variables numéricas en Python, específicamente los tipos int (enteros) y float (decimales).

Aquí tienes un resumen de los puntos clave que cubrimos:

Asignación de variables: Asignamos valores a variables para representar datos, como la cantidad de empleados (Cvigilante, Cdocente, Ccoordinador) y sus respectivos salarios (Svigilante, Sdocente, Scoordinador).
Operaciones aritméticas: Realizamos operaciones aritméticas básicas, como la suma, para calcular la cantidad total de empleados (TotalEmpleados = Cvigilante + Cdocente + Ccoordinador).
Cálculo de diferencias: Calculamos la diferencia entre el salario más alto y el más bajo (diferencia_salario = salario_coordinador - salario_vigilante).
Promedio ponderado: Calculamos el promedio ponderado de los salarios, multiplicando la cantidad de empleados por su respectivo salario, sumando los resultados y dividiendo entre la cantidad total de empleados (promedio_salarios = (Cvigilante * Svigilante + Cdocente * Sdocente + Ccoordinador * Scoordinador) / TotalEmpleados).
Tipos de datos: Verificamos los tipos de datos de las variables utilizando la función type(). Vimos que TotalEmpleados es de tipo int y promedio_salarios es de tipo float.
En resumen, esta clase nos proporcionó una base sólida para manipular datos numéricos en Python y realizar cálculos básicos. ¿Hay algo en particular que te gustaría repasar o tienes alguna pregunta sobre algún punto específico?


##  06 Para saber más: ¿Qué es la documentación?

La documentación de cualquier lenguaje funciona como un conjunto de guías escritas para ayudar a los desarrolladores a comprender y utilizar un lenguaje de programación. Pueden encontrarse en sitios web creados por el equipo que desarrolló ese lenguaje. Como se mencionó en clase, podemos encontrar la documentación de Python en python.org.

Estas documentaciones suelen incluir información sobre sintaxis y estructuras, bibliotecas y herramientas disponibles, funciones, ejemplos de código y otros puntos útiles para ayudar a escribir código eficiente y correcto. Se trata de un recurso valioso tanto para quienes están aprendiendo un nuevo lenguaje de programación como para personas experimentadas que buscan información específica para su código.

Un detalle a tener en cuenta es que la mayoría de las documentaciones están escritas en inglés, lo que puede dificultar para alguien que no tiene un gran conocimiento de ese idioma. Sin embargo, normalmente hay soporte para la traducción al español a través del traductor del navegador.




##  07 ¿Para qué usar la función type()?

La función type() de Python se puede utilizar para verificar los tipos de datos de una variable, lo que puede ser muy útil en Data Science para asegurarse de que los datos se procesen correctamente y para identificar posibles errores. Sabiendo esto, marca la alternativa que indica un ejemplo de uso de la función type.

La función type() puede ser útil para verificar el tipo de dato de una variable y determinar la mejor forma de manipular o procesar los datos. Para trabajar con estadísticas en datos financieros, por ejemplo, podemos verificar si los datos que tenemos son valores numéricos utilizando type.


Esta función devuelve una clase que representa el tipo del objeto pasado como argumento. De esta manera, podemos usar type para comprender el tipo de dato que tenemos y ajustarlo según las necesidades de un análisis.

## 08 Variables de texto

¡Hola! En esta clase, aprendimos sobre la manipulación de cadenas de texto (strings) en Python. Aquí tienes un resumen de los puntos clave:

* 1-Creación de strings: Las strings se crean asignando un texto a una variable utilizando comillas simples o dobles.
* 2-Métodos de strings: Los objetos string tienen métodos que permiten manipular y cambiar el formato del texto. La sintaxis para usar estos métodos es objeto.método().
* 3-Métodos comunes:
* .upper(): Convierte la string a mayúsculas.
* .lower(): Convierte la string a minúsculas.
* .strip(): Elimina los espacios en blanco al principio y al final de la string.
* .replace(valor_antiguo, valor_nuevo): Sustituye todas las ocurrencias del texto antiguo por el nuevo.

* 4-Almacenamiento de cambios: Es importante almacenar las transformaciones en una variable para que los cambios se guarden. Si no se hace, el texto original permanece inalterado.
* 5-IDs de variables: Cada variable ocupa un espacio de memoria con un ID único. Al modificar una variable y reasignarla, puede que ocupe un nuevo espacio de memoria.
* 6-Recolector de basura: Python tiene un recolector de basura que libera la memoria que ya no se está utilizando.
Espero que este resumen te sea útil. ¿Hay algo más en lo que te pueda ayudar o alguna otra pregunta que tengas sobre este tema?

##  09 Para saber más: otros operadores

Durante las clases, aprendimos a sumar, restar, multiplicar y dividir variables numéricas en Python. Pero también existen otros cálculos matemáticos que se pueden realizar mediante operadores aritméticos, como la exponenciación, el módulo y la división entera.

### Exponenciación (**)

Podemos elevar un número a una potencia específica utilizando el operador **. Para obtener este resultado, definimos un valor a la izquierda del operador y la potencia deseada del número a la derecha. Por ejemplo, sabemos que 2 elevado a la 3 es igual a 8 (2 * 2 * 2 = 8). Para realizar este cálculo, escribimos:

In [43]:
2**3

8

El mismo ejemplo se puede hacer con variables:

In [44]:
operador = 2
potencia = 3
operador ** potencia

8

### Módulo (%)

El operador de módulo % puede devolver el residuo de una división entera entre dos números. Recuerda que el residuo de una división es el número que queda cuando la división no es exacta.

Para calcular el módulo de una división, colocamos el dividendo a la izquierda del operador % y el divisor a la derecha. Por ejemplo, para encontrar el residuo de la división de 7 entre 3, podemos ejecutar los siguientes códigos:

In [45]:
7%3

1

O con variables:

In [46]:
dividendo = 7
divisor = 3
dividendo % divisor

1

### División entera (//)

Esta operación devuelve el resultado de la división entera entre dos números, es decir, solo devuelve la parte entera de una división. Por ejemplo, sabemos que la división exacta de 7 por 3 (7/3) resulta en el valor 2.333333.... Si esta fuera una división entera, el resultado sería simplemente 2. La sintaxis consiste en colocar el valor del numerador, el operador // y luego el denominador, como se muestra a continuación:

In [47]:
7 // 3

2

El mismo ejemplo se puede hacer con variables:

In [48]:
numerador = 7
denominador = 3
numerador // denominador

2

###  10 Colectando datos


Puedes probar todos los ejemplos en tu cuaderno, verificar sus resultados y realizar pruebas modificando los valores.

###  11 Para saber más: tabla unicode

Python es un lenguaje de programación caracterizado por el uso del estándar Unicode de caracteres. Este estándar permite que desarrolladores de todo el mundo trabajen con textos y caracteres en diferentes idiomas de manera muy sencilla, sin preocuparse por problemas de compatibilidad.

Esto fue posible gracias al desarrollo de la tabla Unicode, una tabla de codificación de caracteres que asocia códigos numéricos con caracteres específicos. Su objetivo era incluir caracteres de todos los idiomas y sistemas de escritura existentes en el mundo. Hasta la versión 15.0, el estándar ya agrupa más de 140,000 caracteres, que incluyen letras, números, símbolos y emojis. Puedes ver las estadísticas de cada versión y la cantidad de caracteres por versión en este enlace.

Además, Unicode surgió como una alternativa a otras tablas de caracteres, como ASCII, que solo admite caracteres del alfabeto inglés y algunos símbolos. Por lo tanto, es muy importante ya que permite la representación de caracteres de diferentes idiomas (como la "ç" en portugués, la "ñ" en español o la "П" en ruso) y scripts de manera consistente e independiente del dispositivo o plataforma, lo que es fundamental para la globalización y la representación precisa de textos en diferentes idiomas.

Es importante recordar que la tabla Unicode se actualiza y se expande constantemente, incluyendo nuevos caracteres y scripts. Esto asegura que siga siendo relevante y útil para admitir la representación de todos los idiomas y sistemas de escritura del mundo. Puedes seguir las actualizaciones en el sitio unicode.org y acceder a las codificaciones de cada carácter de este estándar en la Wikipedia o en documentos oficiales en PDF.

Podemos imprimir un elemento de la tabla Unicode en nuestro cuaderno utilizando la función chr, que devuelve una cadena con el carácter especificado. Su sintaxis es la siguiente: chr(número_del_carácter). Por ejemplo, vamos a imprimir el carácter "@" con chr(). Según la tabla, sabemos que "@" corresponde al número 64, así que lo reprodujimos:

In [49]:
chr(64)

'@'

Se pueden probar y observar varias combinaciones de números en las salidas. Siéntete libre de probar varias combinaciones e incluso crear frases.

También es posible concatenar cadenas con el operador "+". Sabiendo esto, si unimos varios caracteres, podemos formar una palabra. Por ejemplo, vamos a construir la palabra "Hola" ('H' + 'o' + 'l' + 'a'):

In [50]:
chr(72) + chr(111) +  chr(108) + chr(97)

'Hola'

###  12 Para saber más: diversos formatos de print()

Podemos visualizar el resultado de variables dentro de cadenas de texto, así como imprimir el texto final con un print. Durante la lección, aprendimos a utilizar la f-string (formateo de cadena), en la que colocamos una f antes de crear la cadena y las variables se colocan entre llaves {}. Ejemplo:



In [51]:
nombre = "Ana María"
edad = 17
print(f"El nombre de la alumna es {nombre} y su edad es {edad} años.")

El nombre de la alumna es Ana María y su edad es 17 años.


Pero existen otros métodos de formateo, como el uso del operador de formateo de cadena o la función .format().

Operador de Formateo:

Este operador de formateo permite la inserción de variables en puntos específicos en la cadena de texto utilizando el operador %. Este operador funciona como un marcador, indicando dónde se expondrá el valor de la variable en la cadena.

El % debe ir acompañado de una palabra clave para cada tipo de variable que se desee agregar. De acuerdo con la tabla siguiente:

* Tipo de Variable	Palabra Clave
* Cadena de texto	%s
* Entero	%d
* Punto flotante	%f
* Caractér	%c


De esta manera, para insertar una variable, puedes agregar el operador en la cadena en el punto deseado. Luego del final de la cadena, agrega nuevamente el %, especificando la variable entre paréntesis. Podemos ver esta estructura en el ejemplo siguiente:



In [52]:
nombre_alumno = 'Penélope Camacho'
print('Nombre del alumno: %s' %(nombre_alumno))

Nombre del alumno: Penélope Camacho


Si tienes más de una variable, debes ordenarlas según su aparición en el texto y separarlas por comas. Por ejemplo:

In [53]:
nombre_alumno = 'Penélope Camacho'
edad_alumno = 11
media_alumno = 9.95
print('Nombre del alumno es %s, tiene %d años y su media es %f.' %(nombre_alumno, edad_alumno, media_alumno))

Nombre del alumno es Penélope Camacho, tiene 11 años y su media es 9.950000.


Cuando trabajamos con números de coma flotante (float), podemos determinar la cantidad de decimales después del punto utilizando la sintaxis %.xf, donde x es el número de decimales deseados. Utilizando las mismas variables del ejemplo anterior, el código con %.xf se vería así:

In [54]:
print('Nombre del alumno es %s, tiene %d años y su media es %.2f.' %(nombre_alumno, edad_alumno, media_alumno))

Nombre del alumno es Penélope Camacho, tiene 11 años y su media es 9.95.


Un detalle importante es que los operadores de formateo de cadena con % no funcionan directamente con valores booleanos. Una forma de manejarlo es convirtiendo el valor booleano en una cadena antes de utilizarlo en el formateo con la función str(). Por ejemplo:



In [55]:
x = True
print("Valor de x: %s" % str(x))

Valor de x: True


Esto no es un problema con las f-strings o .format.

Método .format():

También es posible utilizar el método .format() para formatear cadenas. Es más flexible y permite pasar las variables directamente dentro de la cadena, sin necesidad de operadores %. Los marcadores son simplemente {}. Por ejemplo:

In [56]:
nombre_alumno = 'Penélope Camacho'
print('Nombre del alumno: {}'.format(nombre_alumno))

Nombre del alumno: Penélope Camacho


Del mismo modo que se hace con el operador, puedes usarlo con varias variables:

In [57]:
nombre_alumno = 'Fabricio Daniel'
edad_alumno = 15
media_alumno = 9.95
print('Nombre del alumno es {}, tiene {} años y su media es {}.' .format(nombre_alumno, edad_alumno, media_alumno))

Nombre del alumno es Fabricio Daniel, tiene 15 años y su media es 9.95.


Nota que con .format(), no tienes el problema de los decimales de coma flotante. Tampoco tienes este problema con las f-strings.

En resumen, cada uno de estos métodos tiene sus ventajas y desventajas. Se recomienda utilizar f-strings, ya que son más legibles y fáciles de usar. Sin embargo, cada desarrollador puede elegir el método que le parezca más apropiado.

* Caracteres Especiales:

Además de la formateo de inserción de variables en una cadena, también existen caracteres especiales. Se utilizan para representar acciones especiales o caracteres que no se pueden escribir directamente, como el “Enter” y la tabulación. A continuación, presentamos algunos de ellos. Intenta replicar todos los ejemplos y observa el resultado final.

\n es el carácter de nueva línea y se usa para saltar una línea en el texto (similar a la función "Enter"). Ejemplo:

In [58]:
print("Estudiar es un esfuerzo constante,\nEs como cultivar una planta,\nNecesitamos dedicación y paciencia,\nPara ver madurar el fruto.")

Estudiar es un esfuerzo constante,
Es como cultivar una planta,
Necesitamos dedicación y paciencia,
Para ver madurar el fruto.


\t es el carácter de tabulación y se utiliza para agregar un espacio de tabulación en el texto. Ejemplo:

In [59]:
print('Cantidad\tCalidad\n5 muestras\tAlta\n3 muestras\tBaja')

Cantidad	Calidad
5 muestras	Alta
3 muestras	Baja


\ \ se usa para imprimir una sola barra invertida. Si no se utiliza una doble barra invertida, el código podría generar un error o un resultado inesperado, ya que Python considera \ como un carácter especial. Usamos esta sintaxis para garantizar que no ocurran errores. Ejemplo:

In [60]:
print("Ruta del archivo: C:\\archivos\\documento.csv")

Ruta del archivo: C:\archivos\documento.csv


\ " se utiliza para imprimir comillas dobles cuando estamos trabajando con una cadena creada con comillas dobles " en el interior. Sin embargo, esto no es necesario si la cadena se crea con comillas simples '. Ejemplo:

In [61]:
print("Una vez oí: \"Los frutos del conocimiento son los más dulces y duraderos de todos.\"")

Una vez oí: "Los frutos del conocimiento son los más dulces y duraderos de todos."


\ ' se utiliza para imprimir comillas simples cuando estamos trabajando con una cadena creada con comillas simples '. Si la cadena se crea con comillas dobles ", esto no es necesario. Ejemplo:

In [62]:
print('Mi profesora una vez dijo: \'Estudiar es la clave del éxito.\'')

Mi profesora una vez dijo: 'Estudiar es la clave del éxito.'


###  13 ¿Qué hace la función input()?

Aprendemos otra función de Python llamada input(). Selecciona la alternativa que describe lo que hace esta función y proporciona un ejemplo con su sintaxis.

La función input se utiliza para leer las entradas del usuario en formato de cadena (string) y devolver el valor leído. Un ejemplo de uso de esta función es el siguiente código:

nombre = input('¿Cuál es tu nombre? ')

La función input permite que el usuario introduzca datos en el programa. Su sintaxis es sencilla, ya que escribimos "input" seguido de paréntesis (). Dentro de ellos, podemos colocar texto que aparecerá antes de la lectura. Al ejecutar el código de la alternativa, el programa imprimirá el mensaje "¿Cuál es tu nombre?" y esperará la entrada del usuario. Cuando el usuario escriba su nombre y presione la tecla Enter, el programa almacenará la entrada en una variable llamada "nombre".

###  14 Desafío: ¡Es hora de practicar!

Vamos a practicar el uso de varios tipos de variables y la función input a través de algunas actividades. Resuelve los problemas propuestos en código.

Recopilación y muestra de datos

1 - Crea un programa que solicite al usuario que escriba su nombre y luego imprima "Hola, [nombre]."

2 - Crea un programa que solicite al usuario que escriba su nombre y edad, y luego imprima "Hola, [nombre], tienes [edad] años."

3 - Crea un programa que solicite al usuario que escriba su nombre, edad y altura en metros, y luego imprima "Hola, [nombre], tienes [edad] años y mides [altura] metros."

Calculadora con operadores

4 - Crea un programa que solicite dos valores numéricos al usuario y luego imprima la suma de ambos valores.

5 - Crea un programa que solicite tres valores numéricos al usuario y luego imprima la suma de los tres valores.

6 - Crea un programa que solicite dos valores numéricos al usuario y luego imprima la resta del primero menos el segundo valor.

7 - Crea un programa que solicite dos valores numéricos al usuario y luego imprima la multiplicación de los dos valores.

8 - Crea un programa que solicite dos valores numéricos, un numerador y un denominador, y realice la división entre los dos valores. Asegúrate de que el valor del denominador no sea igual a 0.

9 - Crea un programa que solicite dos valores numéricos, un operador y una potencia, y realice la exponenciación entre estos dos valores.

10 - Crea un programa que solicite dos valores numéricos, un numerador y un denominador, y realice la división entera entre los dos valores. Asegúrate de que el valor del denominador no sea igual a 0.

11 - Crea un programa que solicite dos valores numéricos, un numerador y un denominador, y devuelva el resto de la división entre los dos valores. Asegúrate de que el valor del denominador no sea igual a 0.

12 - Crea un código que solicite las 3 notas de un estudiante e imprima el promedio de las notas.

13 - Crea un código que calcule e imprima el promedio ponderado de los números 5, 12, 20 y 15 con pesos respectivamente iguales a 1, 2, 3 y 4.

Editando textos

14 - Crea una variable llamada "frase" y asígnale una cadena de texto de tu elección. Luego, imprime la frase en pantalla.

15 - Crea un código que solicite una frase y luego imprima la frase en pantalla.

16 - Crea un código que solicite una frase al usuario y luego imprima la misma frase ingresada pero en mayúsculas.

17 - Crea un código que solicite una frase al usuario y luego imprima la misma frase ingresada pero en minúsculas.

18 - Crea una variable llamada "frase" y asígnale una cadena de texto de tu elección. Luego, imprime la frase sin espacios en blanco al principio y al final.

19 - Crea un código que solicite una frase al usuario y luego imprima la misma frase sin espacios en blanco al principio y al final.

20 - Crea un código que solicite una frase al usuario y luego imprima la misma frase sin espacios en blanco al principio y al final, además de convertirla a minúsculas.

21 - Crea un código que solicite una frase al usuario y luego imprima la misma frase con todas las vocales "e" reemplazadas por la letra "f".

22 - Crea un código que solicite una frase al usuario y luego imprima la misma frase con todas las vocales "a" reemplazadas por el carácter "@".

23 - Crea un código que solicite una frase al usuario y luego imprima la misma frase con todas las consonantes "s" reemplazadas por el carácter "$".

Si necesitas ayuda, las opciones de solución a las actividades estarán disponibles en la sección "Opinión del instructor".



### Opinión del instructor

Pueden existir diversas formas de resolver una cuestión. A continuación, presentaré cómo resolví los problemas, lo que no significa que sean las mejores soluciones, pero son una opción de solución.

Recopilación y muestreo de datos

1 - Podemos solicitar el nombre con la función input y asignar el resultado de la salida a una variable. A continuación, podemos imprimir el resultado de la variable en la función print usando la formatación f-string.

In [1]:
nombre = input('Ingrese su nombre: ')
print(f'¡Hola, {nombre}!')

¡Hola, pool!


2 - Podemos solicitar el nombre y la edad con la función input y asignar el resultado de la salida a una variable. En el caso de la edad, es necesaria una conversión de la salida de input a un valor entero con la función int(). A continuación, podemos imprimir el resultado de las recopilaciones en la función print usando la formateación f-string.

In [2]:
nombre = input('Ingrese su nombre: ')
edad = int(input('Ingrese su edad: '))
print(f'¡Hola, {nombre}, tienes {edad} años!')

¡Hola, pol, tienes 90 años!


3 - Podemos solicitar el nombre, la edad y la altura con la función input y asignar el resultado de la salida a una variable. En el caso de la edad, es necesaria una conversión de la salida de input a un valor entero con la función int(). Para la altura, es necesario realizar una conversión al tipo de valor flotante con la función float(). A continuación, podemos imprimir el resultado de las recopilaciones en la función print usando la formateación f-string.

In [65]:
nombre = input('Ingrese su nombre: ')
edad = int(input('Ingrese su edad: '))
altura = float(input('Ingrese su altura: '))
print(f'¡Hola, {nombre}, tienes {edad} años y mides {altura} metros!')

¡Hola, pol, tienes 99 años y mides 1.8 metros!


Calculadora con operadores

4 - Podemos recopilar los dos valores con input y convertir la salida a un número entero con la función int(). Luego, podemos mostrar el resultado de la suma entre las dos entradas con +.



In [3]:
a = int(input('Ingrese el primer valor: '))
b = int(input('Ingrese el segundo valor: '))
print(a + b)

17


5 - Usando la misma lógica que en la pregunta anterior, podemos recopilar los tres valores con input y convertir la salida a números enteros con la función int(). Luego, podemos mostrar el resultado de la suma de los tres valores con +.

In [67]:
a = int(input('Ingrese el primer valor: '))
b = int(input('Ingrese el segundo valor: '))
c = int(input('Ingrese el tercer valor: '))
print(a + b + c)

22


6 - Podemos recopilar los dos valores con input y convertir la salida a números enteros con la función int(). Luego, podemos mostrar el resultado de la resta del primer valor menos el segundo valor con -.

In [68]:
a = int(input('Ingrese el primer valor: '))
b = int(input('Ingrese el segundo valor: '))
print(a - b)

-9


7 - Podemos recopilar los dos valores con input y convertir la salida a números enteros con la función int(). Luego, podemos mostrar el resultado de la multiplicación de los dos valores con *.

In [69]:
a = int(input('Ingrese el primer valor: '))
b = int(input('Ingrese el segundo valor: '))
print(a * b)

5829


8 - Podemos recopilar el numerador y denominador con input y convertir la salida a números enteros con la función int(). Luego, podemos mostrar el resultado de la división entre los dos valores con /.



In [70]:
numerador = int(input('Ingrese el numerador: '))
denominador = int(input('Ingrese el denominador (OBS: el valor no puede ser nulo): '))
print(numerador / denominador)

0.09210526315789473


9 - Podemos recopilar el operador y potencia con input y convertir la salida a números enteros con la función int(). Luego, podemos mostrar el resultado de la exponenciación entre estos dos valores con **.

In [71]:
operador = int(input('Ingrese el valor del operador: '))
potencia = int(input('Ingrese el valor de la potencia: '))
print(operador ** potencia)

387420489


10 - Podemos recopilar el numerador y denominador con input y convertir la salida a números enteros con la función int(). Luego, podemos mostrar el resultado de la división entera entre los dos valores con //.

In [72]:
numerador = int(input('Ingrese el numerador: '))
denominador = int(input('Ingrese el denominador (OBS: el valor no puede ser nulo): '))
print(numerador // denominador)

9


11 - Podemos recopilar el numerador y denominador con input y convertir la salida a números enteros con la función int(). Luego, podemos mostrar el resultado del resto de la división entre los dos valores con %.

In [73]:
numerador = int(input('Ingrese el numerador: '))
denominador = int(input('Ingrese el denominador (OBS: el valor no puede ser nulo): '))
print(numerador % denominador)

1


12 - Podemos recopilar las tres notas con input y convertir la salida a números con decimales utilizando la función float(). Luego, podemos mostrar el resultado del promedio entre las tres variables sumando las notas con + y dividiendo la suma entre 3.

In [74]:
nota_1 = float(input('Ingrese la 1° nota: '))
nota_2 = float(input('Ingrese la 2° nota: '))
nota_3 = float(input('Ingrese la 3° nota: '))
print(f'Media {(nota_1+nota_2+nota_3)/3}.')

Media 6.7.


13 - Calculamos la media ponderada multiplicando los pesos por sus respectivos valores y sumando el resultado de cada multiplicación. El valor de estas sumas se divide entre la suma total de los pesos. Con el comando print, imprimimos el resultado del cálculo de la media.

In [75]:
media_ponderada = (5*1 + 12*2 + 20*3 + 15*4) / (1+2+3+4)
print(f'Media {media_ponderada}.')

Media 14.9.


Editando textos

14 - Definimos una frase arbitraria entre comillas simples y la imprimimos insertando la variable en un comando print.

In [76]:
frase = '¡Hola Python!'
print(frase)

¡Hola Python!


15 - Recopilamos una frase utilizando la función input y luego imprimimos el resultado con el comando print.

In [77]:
frase = input('Escribe una frase: ')
print(frase)

vamos pol


16 - Recopilamos una frase utilizando la función input y luego mostramos su valor en mayúsculas utilizando el método upper. Podemos mostrar la salida de esto en un comando print.

In [78]:
frase = input('Escribe una frase: ')
print(frase.upper())

POLITO ES COOL


17 - Recopilamos una frase utilizando la función input y luego mostramos su valor en minúsculas utilizando el método lower. Podemos mostrar la salida de esto en un comando print.

In [79]:
frase = input('Escribe una frase: ')
print(frase.lower())

polll


18 - Para este ejemplo, es interesante definir una frase con espacios al principio y al final de la frase. Una vez hecho esto, podemos eliminar estos espacios utilizando el método strip y mostrar el resultado en un comando print.

In [80]:
frase = ' ¡Hola Python! '
print(frase.strip())

¡Hola Python!


19 - Recopilamos una frase utilizando la función input, incluso si no estamos seguros de si tendrá espacios al principio y al final de la frase. Luego, podemos eliminar estos espacios utilizando el método strip. El resultado se puede mostrar en un comando print.

In [81]:
frase = input('Escribe una frase: ')
print(frase.strip())

viva yo


20 - Recopilamos una frase utilizando la función input, incluso si no estamos seguros de si tendrá espacios al principio y al final de la frase. Luego, eliminamos estos espacios utilizando el método strip y también usamos el método lower junto con strip. El resultado se puede mostrar en un comando print.

In [82]:
frase = input('Escribe una frase: ')
print(frase.strip().lower())

pol es genial


21 - Recopilamos una frase utilizando la función input. Para asegurarnos de que los caracteres no estén en mayúsculas, convertimos toda la frase a minúsculas utilizando el método lower y luego aplicamos el método replace, definiendo el valor a reemplazar como 'e' y el nuevo valor como 'f'. El resultado se muestra en un comando print.

In [83]:
frase = input('Escribe una frase: ')
print(frase.lower().replace('e','f'))

polito fs cool


22 - Recopilamos una frase utilizando la función input. Para asegurarnos de que los caracteres no estén en mayúsculas, convertimos toda la frase a minúsculas utilizando el método lower y luego aplicamos el método replace, definiendo el valor a reemplazar como 'a' y el nuevo valor como el carácter 64 según la tabla Unicode, que corresponde al carácter @. El resultado se muestra en un comando print.



In [84]:
frase = input('Escribe una frase: ')
print(frase.lower().replace('a',chr(64)))

pol


23 - Recopilamos una frase utilizando la función input. Para asegurarnos de que los caracteres no estén en mayúsculas, convertimos toda la frase a minúsculas utilizando el método lower y luego aplicamos el método replace, definiendo el valor a reemplazar como 's' y el nuevo valor como el carácter 36 según la tabla Unicode, que corresponde al símbolo $. El resultado se muestra en un comando print.



In [85]:
frase = input('Escribe una frase: ')
print(frase.lower().replace('s',chr(36)))

libre como el viento


Ejercitarse es importante para consolidar el contenido, desarrollar habilidades de codificación, identificar puntos que aún no se han comprendido, prepararse para desafíos futuros, desarrollar el pensamiento lógico y la resolución de problemas.

Teniendo esto en cuenta, es interesante realizar muchos ejercicios cuando se está aprendiendo un nuevo lenguaje de programación. Por lo tanto, busca completar los ejercicios propuestos y observa cómo te ayudan a progresar en tus habilidades de codificación.

Si tienes alguna pregunta, puedes comunicarte en el foro.

## Manipulando los datos
Vamos a aprender sobre las variables en Python, cómo declararlas y utilizarlas además de conocer otros comandos de este lenguaje. :D

# Variables

In [86]:
7

7

In [87]:
id(7)

140101310005680

In [88]:
edad = 7

In [89]:


print(id(edad))
print(id(7))

140101310005680
140101310005680


In [90]:
numerosiete = 7
id(numerosiete)

140101310005680

In [91]:
nombre = 'Pablo'
id('Pablo')


140101102756016

In [92]:
id(nombre)

140101102756016

# Existen algunos tipos de nombres que no podemos definir para las variables:

* Nombres que comiencen con números
* Ejemplos: 10_notas, 2_nombre_casa, etc.
* Palabras separadas por espacio
* Ejemplos: Nombre escuela, notas estudiantes, etc.
* Nombres de funciones de Python
* Ejemplos: print, type, etc.

In [93]:
edad = 1
Edad = 2
EDAD = 3
_edad = 4
_edad_ = 5
print(edad, Edad, EDAD, _edad, _edad_)


1 2 3 4 5


In [94]:
edad = 7
type(edad)

int

In [95]:
promedio = 9.7
type(promedio)

float

In [96]:
nombre = "Pablo"
type(nombre)

str

In [97]:

v = True
type(v)

bool

En un conjunto de datos escolares podemos tener varios tipos de informaciones. Supongamos que tenemos acceso a la ficha de datos del alumno Pablo Monsalvo, cómo transformamos este nombre en variables de Python?

Ficha:
* Nombre: Pablo Monsalvo
* Edad: 44 años
* Promedio del semestre: 9,75
* Situación de aprobación: Verdadera (aprobado)

In [98]:
nombre_estudiante = 'Pablo Monsalvo'
edad_estudiante = 44
promedio_estudiante = 9.75
aprobado = True

print(nombre_estudiante,edad_estudiante,promedio_estudiante,aprobado)
print(type(nombre_estudiante),type(edad_estudiante),type(promedio_estudiante),type(aprobado))

Pablo Monsalvo 44 9.75 True
<class 'str'> <class 'int'> <class 'float'> <class 'bool'>


# Variables numéricas

Entre los tipos de datos numéricos vamos a enfocarnos en los tipos entero y float.

Tenemos una tabla de información sobre los diversos cargos, cantidad de personas empleadas y el salario correspondiente:

* Cargo	     Cantidad  	Salario
* Vigilante	    5	       300
* Docente	      16	     500
* Coordinador	  2	       600
* Necesitamos trabajar con estos datos para obtener:

* La cantidad total de empleados;
* La diferencia entre el salario más bajo y más alto; y
* El promedio ponderado de los salarios de los empleados de la escuela.


In [99]:

c_vigilante = 5
s_vigilante = 300

c_docente = 16
s_docente = 500

c_coordinador = 2
s_coordinador = 600

In [100]:

total_empleados = c_vigilante + c_docente + c_coordinador
total_empleados

23

In [101]:
diferencia_salario = s_coordinador - s_vigilante
diferencia_salario


300

In [102]:
type(diferencia_salario)

int

In [103]:
promedio_salarios = (c_vigilante * s_vigilante + c_docente * s_docente
                     + c_coordinador * s_coordinador) / total_empleados
promedio_salarios

465.2173913043478

In [104]:
type(promedio_salarios)

float

# Strings
Strings hace referencia a un conjunto de caracteres formando un texto.

Podemos crear Strings cuando atribuimos un dato a una variable entre comillas sencillas (') o comillas dobles (").

In [105]:
t = 'Alura'
type(t)


str

Las variables textuales son objetos que poseen métodos que nos ayudan a formatear strings. Los métodos los podemos ejecutar al definir un objeto de acuerdo con la siguiente estructura:

# objeto.metodo()
Existen métodos que no necesitan los (), siempre es una buena práctica verificar la documentación para cada caso.

# Situación:

Recibimos una variable con el nombre de un profesor de la escuela para añadirla a los registros. Sin embargo, necesitamos darle un tratamiento a este texto antes de insertarlo al sistema.

In [106]:
texto = ' Pablo Matías Monsalvo '
id(texto)

140101102769792

El objetivo final es que el nombre aparezca de la siguiente forma:

'PABLO MATIAS MONSLVO'

# Convierte una string a mayúsculas.

In [107]:
texto.upper()

' PABLO MATÍAS MONSALVO '

# Convierte una string a minúsculas.

In [108]:
texto.lower()

' pablo matías monsalvo '

# Elimina los espacios en blanco al inicio y al final de una string.

In [109]:
texto.strip()

'Pablo Matías Monsalvo'

# Sustituye todas las ocurrencias del texto "antiguo" en la string por el "nuevo".

In [110]:
texto.replace('s','z')

' Pablo Matíaz Monzalvo '

# Observaciones
Los métodos devuelven una tranformación, no son ejecutados directamente sobre la variable donde está almacenado el texto.

Adicionalmente, podemos acumular la ejecución de los métodos.

In [111]:
texto

' Pablo Matías Monsalvo '

In [112]:
id(texto)

140101102769792

# Para que la transformación se pueda ejecutar debemos atribuir la salida de las transformaciones a las variables

In [113]:
nuevo_texto = texto.strip().replace('s','z').upper()
print(nuevo_texto)
print(id(texto), id(nuevo_texto))

PABLO MATÍAZ MONZALVO
140101102769792 140101102682000


In [114]:
texto = texto.strip().replace('s','z').upper()
print(texto)
print(id(texto), id(nuevo_texto))

PABLO MATÍAZ MONZALVO
140101102679408 140101102682000


# Capturando datos
En algunas aplicaciones necesitamos capturar los valores del usuario de nuestro proyecto. En python, logramos capturar los datos del usuario a través del comando input().

Para ejecutar la captura debemos atribuir el resultado de esta función a una variable.

In [115]:
nombre = input("Escribe tu nombre: ")

In [116]:
type(nombre)

str

La salida de este comando siempre será una string. Esto quiere decir que aunque capturemos un valor de variable numérica, este será de tipo string.

Entonces, será necesario convertir el resultado cuando no se desea utilizar el valor como tipo string.

Existen funciones para la conversión de valores:

* Enteros: int(dato_para_conversion)
* Coma flotante: float(dato_para_conversion)
* String: str(dato_para_conversion)
* Booleano: bool(dato_para_conversion)

In [117]:

año_admision = int(input('Digita el año de admisión: '))

In [118]:
type(año_admision)


int

In [119]:
nota_admision = float(input('Digita la nota de admisión: '))


In [120]:
type(nota_admision)

float

Trataremos de representar mejor ahora el resultado de la transformación que obtuvimos. Nosotros podemos formatear y presentar nuestro resultado mezclando strings con valores no textuales.

Para ello utilizamos la estructura de formatación f con strings.

In [121]:
print(f'\tEl año de admisión fue: {año_admision} \n\tLa nota de admisión fue: {nota_admision}')


	El año de admisión fue: 2025 
	La nota de admisión fue: 9.9


# Estructuras condicionales

In [122]:
# Nota de aprobación entre 7 y 10 == nota de aprobación >= 7
# Nota de reprobación es menor a 7 == nota de reprobación < 7

"""
Si la nota es mayor o igual a 7 entonces el estudiante aprueba
Si no, entonces el estudiante reprueba
"""

'\nSi la nota es mayor o igual a 7 entonces el estudiante aprueba\nSi no, entonces el estudiante reprueba\n'

# IF y ELSE

Los comandos if y else son dos estructuras condicionales. El comando if ejecutará el bloque de comandos en caso de que la condición citada sea verdadera. El comando else ejecutará el bloque de comandos en el caso de que la condicional de if sea falsa.

El comando if es una palabra clave en Python que significa "si". Este se emplea para conformar la estructura condicional, que te permite verificar si una determinada condición es verdadera o falsa e, inmediatamente después, ejecute un bloque de código específico dependiendo del resultado de la verificación. La sintaxis para usar el comando if es:

if condicion:
    # Realiza algo

In [123]:
if 2 < 7:
  print('La condición es verdadera.')

La condición es verdadera.


In [124]:
if 2 > 7:
  print('La condición es verdadera.')
print('Fin de la consulta.')


Fin de la consulta.


Ya el comando else en Python se usa en junción con la palabra clave if para formar una estructura condicional. la sintaxis para usar el else es:

if condicion:
  # codigo en caso de que sea verdad
else:
  # codigo en caso de que sea falso
el comando else se ejecuta cuando la condición verificada por el if es evaluada como False.

In [125]:
if 2 > 7:
  print('La condición es verdadera.')
else:
  print('La condición es falsa.')
print('Fin de la consulta.')

La condición es falsa.
Fin de la consulta.


# Situación:

Recibiremos el promedio de la nota de los estudiantes y necesitamos de un algoritmo que ejecute el análisis y determine si el estudiante fue Aprobado o Reprobado, mostrando un mensaje del resultado. Para ser aprobado, el promedio necesita ser igual o superior a 7.0.

In [126]:
nota = float(input('Digita la nota: '))

if nota >= 7:
  print('Aprobó.')
else:
  print('Reprobó.')

Aprobó.


Ahora, nuestra institución educativa estableció que las personas que tengan el promedio entre 5.0 y 7.0 pueden participar del curso de Recuperación durante las vacaciones para lograr aprobar.

Entonces podemos apoyarnos en un conjunto de ifs para poder estructurar esta nueva condición.

In [127]:
nota = float(input('Digita la nota: '))

if nota >= 7:
  print('Aprobó.')
if 7 > nota >= 5:
  print('Recuperación.')
if nota < 5:
  print('Reprobó.')


Aprobó.


Observa que en casos con 3 situaciones como este necesitamos definir bien nuestras condiciones. Pues, si realizamos una condición con else al final, este irá a considerar solamente la condición if inmediatamente anterior para generar la salida en caso de que el resultado de su operación lógica sea falso dando como resultado, dos o más ejecuciones.

Por ejemplo:

In [128]:
nota = float(input('Digita la nota: '))

if nota >= 7:
  print('Aprobó.')
if 7 > nota >= 5:
  print('Recuperación.')
else:
  print('Reprobó.')

Aprobó.
Reprobó.


# ELIF
El comando elif es una palabra clave en Python que significa "si no, si" y lo podemos considerar una contracción entre else e if. Se utiliza en conjunto con la palabra clave if para formar una estructura condicional en cadena.

la sintaxis para utilizar el comando elif es:

if condicion1:
    # Realiza algo
elif condicion2:
    # Realiza otra cosa
elif condicion3:
    # Realiza otra cosa
else:
    # Realiza algo diferente

El comando elif permite que puedas verificar varias condiciones en cadena, economizando espacio en tu código. Si la primera condición se evalúa como False, el interpretador de Python evaluará la próxima condición con el elif. Ello continuará hasta que una condición sea evaluada como True o hasta que el else sea alcazado. Si ninguna de las condiciones es evaluada como True, la ejecución del código del comando else será iniciada.

Vamos a emplear el mismo caso anterior:

In [129]:

nota = float(input('Digita la nota: '))

if nota >= 7:
  print('Aprobó.')
elif 7 > nota >= 5:
  print('Recuperación.')
else:
  print('Reprobó.')

ValueError: could not convert string to float: '9-9'

# Operadores
Durante la construcción de comandos a veces necesitamos de una elaboración mayor de la expresión condicional, necesitando que algunos operadores lógicos se encuentren integrados.

# AND, OR, NOT
Los operadores lógicos and, or y not son usados para combinar expresiones lógicas en Python. Ellos son usados frecuentemente en conjunto con el comando if para crear estructuras condicionales más complejas.

* AND es usado para verificar si dos condiciones son verdaderas. La expresión lógica¹ x and y se evalúa como True tan solo si ambas condiciones x y y son verdaderas, y como False en caso contrario.

* OR es usado para verificar si al menos una de las condiciones es verdadera. La expresión lógica x or y se evalúa como True si al menos una de las condiciones x o y es verdadera, y como False si ambas condiciones son falsas.

* NOT es usado para negar una condición. La expresión lógica not x es evaluada como True si la condición x es falsa, y como False si la condición x es verdadera.

¹ Una expresión lógica es una declaración que puede ser evaluada como verdadera o falsa. Ella se compone por operandos lógicos² y por operadores lógicos³, que son usados ​​para combinar varias expresiones lógicas en una única expresión.

² Los operandos lógicos son los elementos que son comparados o evaluados en una expresión lógica. Ellos son generalmente valores verdaderos o falsos, pero también pueden ser expresiones lógicas más complejas. En Python, los operandos lógicos son los valores True y False.

³ Los operadores lógicos son ls símbolos o palabras clave que son usados ​​para combinar varias expresiones lógicas en una única expresión. En Python, los operadores lógicos son and, or y not, bien como las palabras clave if, elif e else.

Operadores lógicos más comunes

a	b	AND	OR	NOT
|0|0|0|0|1 |0|1|0|1|- |1|0|0|1|- |1|1|1|1|0

In [ ]:

v = True
f = False

In [ ]:
if f and f:
  print('La condición es verdadera.')
else:
  print('La condición es falsa.')

La condición es falsa.


In [ ]:
if v or v:
  print('La condición es verdadera.')
else:
  print('La condición es falsa.')


La condición es verdadera.


In [ ]:

if not f:
  print('La condición es verdadera.')
else:
  print('La condición es falsa.')


La condición es verdadera.


# IN
Es usado para verificar si un elemento está presente en una lista, tupla u otra colección de conjunto. La expresión x in y se evalúa como True si el elemento x esta presente  la variable de conjunto y, y como False en caso contrario.

Podemos verificarlo con variables de texto.



# Situación:

En la escuela nos entregaron una lista con nombres de estudiantes que fueron aprobados según el promedio del semestre, pero es necesario verificar si algunos nombres están en esa lista para confirmar que los datos estén correctos.

La lista distribuida puede ser observada a continuación:

lista = 'Juan Pérez, María González, Pedro Rodríguez, Ana López, Carlos Martínez,
    Laura Sánchez, José García, Elena Fernández, Luis Morales, Carmen Torres,
    David Ruiz, Isabel Ramírez, Javier Díaz, Sara Herrera, Miguel Castro,
    Patricia Ortega, Francisco Vargas, Marta Jiménez, Manuel Medina, Rosa Molina,
    Alejandro Silva, Silvia Ruiz, Andrés Torres, Natalia Soto, Diego Guerrero,
    Paula Ríos, Ricardo Navarro, Alicia Cordero, Carlos Vidal, Lorena Gómez'
Los nombres que necesitas verificar son los siguientes:

nombre_1 = 'Miguel Castro'
nombre_2 = 'Marcelo Noguera

In [ ]:
lista = 'Juan Pérez, María González, Pedro Rodríguez,Ana López, Carlos Martínez,Laura Sánchez, José García, Elena Fernández, Luis Morales, Carmen Torres,David Ruiz, Isabel Ramírez, Javier Díaz, Sara Herrera, Miguel Castro,Patricia Ortega, Francisco Vargas, Marta Jiménez, Manuel Medina, Rosa Molina,Alejandro Silva, Silvia Ruiz, Andrés Torres, Natalia Soto, Diego Guerrero,Paula Ríos, Ricardo Navarro, Alicia Cordero, Carlos Vidal, Lorena Gómez'


In [ ]:
nombre_1 = 'Miguel Castro'
nombre_2 = 'Marcelo Noguera'


In [ ]:
if nombre_1 in lista:
  print('El estudiante aprobó.')
else:
  print('El estudiante reprobó.')

El estudiante aprobó.


In [ ]:

if nombre_2 in lista:
  print('El estudiante aprobó.')
else:
  print('El estudiante reprobó.')

El estudiante reprobó.


# Estructuras de repetición
Cuando tenemos que ejecutar un mismo bloque de comandos durante varias ocasiones no es nada práctico hacerlo a mano.

Situación: Nos fue solicitado obtener e imprimir el promedio de 2 notas de 3 estudantes:

In [ ]:

nota_1 = float(input('Digita la primera nota: '))
nota_2 = float(input('Digita la segunda nota: '))
print(f'El promedio del estudiante 1 es: {(nota_1+nota_2)/2}')

nota_1 = float(input('Digita la primera nota: '))
nota_2 = float(input('Digita la segunda nota: '))
print(f'El promedio del estudiante 2 es: {(nota_1+nota_2)/2}')

nota_1 = float(input('Digita la primera nota: '))
nota_2 = float(input('Digita la segunda nota: '))
print(f'El promedio del estudiante 3 es: {(nota_1+nota_2)/2}')


Digita la primera nota: 4
Digita la segunda nota: 5
El promedio del estudiante 1 es: 4.5
Digita la primera nota: 9
Digita la segunda nota: 8
El promedio del estudiante 2 es: 8.5
Digita la primera nota: 6
Digita la segunda nota: 6
El promedio del estudiante 3 es: 6.0


Ahora imaginemos una situación en que no son tan solo 3 estudiantes, sino 100 estudiantes. No sería nada interesante repetir el mismo código 100 veces, sino ejecutar el mismo código 100 veces.

¡Esta estructura la logramos construir con lazos de repetición!

# WHILE
El lazo while es una estructura de control de repetición en Python que permite ejecutar un bloque de código repetidamente mientras que una determinada condición sea verdadera. Su estructura es:

while condicion:
    # bloque de codigo
Vamos a construir un ejemplo con un contador de 1 hasta 10.

In [ ]:
contador = 1
while contador <= 10:
  print(contador)
  # contador = contador + 1
  contador += 1

1
2
3
4
5
6
7
8
9
10


Ahora vamos a colectar las notas y promedios de cada alumno dentro del lazo while. Haremos un ejemplo con 3 promedios.

In [ ]:
contador = 1
while contador <= 3:
  nota_1 = float(input('Digita la primera nota: '))
  nota_2 = float(input('Digita la segunda nota: '))
  print(f'El promedio del estudiante 1 es: {(nota_1+nota_2)/2}')
  contador += 1

Digita la primera nota: 4
Digita la segunda nota: 5
El promedio del estudiante 1 es: 4.5
Digita la primera nota: 6
Digita la segunda nota: 6
El promedio del estudiante 1 es: 6.0
Digita la primera nota: 8
Digita la segunda nota: 9
El promedio del estudiante 1 es: 8.5


# FOR
El lazo for es un tipo de estructura de control de flujo en Python que permite iterar sobre un conjunto de elementos. Su estructura es:

for elemento in conjunto:
    # codigo a ser ejecutado para cada elemento
El lazo for itera sobre cada elemento del conjunto especificado y ejecuta el bloque de código dentro del lazo para cada elemento. Cuando el lazo llega al final del conjunto, este se interrumpe y el programa continúa la ejecución después del lazo.

El conjunto lo podemos generar con la función incorporada range(). Se trata de una función capaz de generar una secuencia de números enteros. La estructura de esta función es:

range(inicio, fin, paso)
Según la documentación, range() genera una secuencia de números enteros a partir del valor del parámetro inicio hasta el valor del parámetro fin, de acuerdo con el valor del parámetro paso. Si inicio no es especificado, el valor por defecto es 0. Si paso no es especificado, el valor por defecto es 1.

Vamos a recrear el mismo contador while con el lazo for.

In [ ]:

for n in range(1,11):
  print(n)

1
2
3
4
5
6
7
8
9
10


In [ ]:
for e in range(1,4):
  nota_1 = float(input('Digita la primera nota: '))
  nota_2 = float(input('Digita la segunda nota: '))
  print(f'El promedio del estudiante 1 es: {(nota_1+nota_2)/2}')

Digita la primera nota: 4
Digita la segunda nota: 5
El promedio del estudiante 1 es: 4.5
Digita la primera nota: 6
Digita la segunda nota: 6
El promedio del estudiante 1 es: 6.0
Digita la primera nota: 8
Digita la segunda nota: 9
El promedio del estudiante 1 es: 8.5


# Estructuras de datos
Un conjunto de elementos es una colección de items, que son almacenados juntos de manera organizada. Algunos ejemplos de conjuntos de elementos en Python son las listas, los strings y los diccionarios.

# Listas
Las listas pueden almacenar una colección de items en orden. Ellas son delimitadas por corchetes [] y sus elementos son separados por comas.

Estas pueden almacenar cualquier tipo de items, incluyendo números, strings, objetos, otras listas, entre otros. También nos permiten almacenar items de tipos de datos diferentes juntos en una única lista.

In [ ]:
lista = ['Pablo Monsalvo',9.5,9.0,9.7,True]
lista

['Pablo Monsalvo', 9.5, 9.0, 9.7, True]

Las listas son organizadas en Python porque cada elemento de la lista tiene un índice que indica su posición al interior de la lista. Los índices comienzan en 0 van hasta el tamaño de la lista menos 1.

Tenemos entonces 5 elementos con índices que varían de 0 a 4, ordenadamente:

#             [0]           [1]   [2]   [3]    [4]
lista = ['Pablo Monsalvo', 9.5 , 9.0 , 9.7 , True]
En Python tenemos también los índices negativos que se inician en el último elemento con el valor de -1 y avanzan en el universo de los negativos hasta llegar al 1° elemento:

#             [-5]         [-4]  [-3]  [-2]   [-1]
lista = ['Pablo Monsalvo', 9.5 , 9.0 , 9.7 , True]
Logramos seleccionar separadamente cada elemento a través de sus respectivos índices. Colocando el nombre de la lista y en seguida el índice que será seleccionado.

In [ ]:
type(lista[0])


str

In [ ]:
type(lista[1])


float

In [ ]:
type(lista[-1])


bool

Una forma más dinámica de trabajar item a item en una lista es utilizando un lazo for para leer un elemento a la vez.

In [ ]:
for elemento in lista:
  print(elemento)


Pablo Monsalvo
9.5
9.0
9.7
True


Situación: La nota 9.0 de Pablo necesita ser ajustada pues ganó 1 punto en su última nota por participación en clase. Entonces es necesario realizar un cambio en el índice 2 de 9.0 a 10.0

In [ ]:
lista[2] = 10.0


In [ ]:
lista

['Pablo Monsalvo', 9.5, 10.0, 9.7, True]

También podemos calcular el promedio del estudiante a partir de los datos que tenemos.




In [ ]:
promedio = (lista[1]+lista[2]+lista[3])/3
promedio

9.733333333333333

# Manipulación de listas
Las listas son muy útiles en Python porque nos permiten almacenar y acceder a una colección de items de manera organizada y rápida. Estas también ofrecen muchos métodos útiles para manipular los itens almacenados, como adicionar, remover, clasificar y buscar elementos.

# Cantidad de elementos
Usamos la función len() para descubrir la cantidad de elementos de un conjunto.

In [ ]:

len(lista)

5

# Partición
La partición de listas por indexación en Python es una técnica muy útil para seleccionar un subconjunto de elementos de una lista. Esta se realiza usando la sintaxis lista[inicio:fin], donde inicio es el índice del primer elemento que será incluído en la partición fin es el índice del primer elemento que será excluído de la partición.

In [ ]:
lista[0:2]


['Pablo Monsalvo', 9.5]

In [ ]:
lista[:2]


['Pablo Monsalvo', 9.5]

In [ ]:
lista[3:]

[9.7, True]

In [ ]:
lista[:]

['Pablo Monsalvo', 9.5, 10.0, 9.7, True]

# Añade un elemento al final de la lista.

In [ ]:
lista.append(promedio)


In [ ]:
lista

['Pablo Monsalvo', 9.5, 10.0, 9.7, True, 9.733333333333333]

# extend()
Añade varios elementos al final de la lista.

Añadiremos las notas [10.0,8.0,9.0] En la lista de Pablo Monsalvo.

In [ ]:
lista.extend([10.0,8.0,9.0])


In [ ]:
lista

['Pablo Monsalvo', 9.5, 10.0, 9.7, True, 9.733333333333333, 10.0, 8.0, 9.0]

A continuación te mostraremos lo que NO SE PUEDE realizarse con append.

In [ ]:

lista.append([10.0,8.0,9.0])

In [ ]:
lista

['Pablo Monsalvo',
 9.5,
 10.0,
 9.7,
 True,
 9.733333333333333,
 10.0,
 8.0,
 9.0,
 [10.0, 8.0, 9.0]]

# Remueve un elemento específico de la lista.



In [ ]:
lista.remove([10.0, 8.0, 9.0])

In [ ]:
lista

['Pablo Monsalvo', 9.5, 10.0, 9.7, True, 9.733333333333333, 10.0, 8.0, 9.0]

# Diccionario
Los diccionarios son un tipo de estructura de datos que almacenan pares de llave-valor. Estos son delimitados por llaves {} y los pares llave-valor son separados por comas.

diccionario = {llave: valor}
La llave es un elemento único que identifica a un valor en el diccionario, mientras que el valor es el item que será almacenado para la llave. Las llaves y sus respectivos valores se pueden emplear para cualquier tipo de dato.

Los diccionarios son útiles para almacenar y acceder a los datos de manera organizada y rápida. Se trata de un tipo de conjunto de elementos en Python, pues almacenan una colección de items.

In [ ]:
diccionario = {'llave_1':1,
               'llave_2':2}

diccionario


{'llave_1': 1, 'llave_2': 2}

In [ ]:
type(diccionario)

dict

# Situación:

Vamos a crear un conjunto de datos con las informaciones de matrícula de un estudiante. Los datos son los siguientes:

matrícula: 2000168933
día de registro: 25
mes de registro: 10
grupo: 2E

In [ ]:
estudiante = {
              'matricula': 2000168933,
              'dia_registro': 25,
              'mes_registro': 10,
              'grupo': '2E'
              }

estudiante

{'matricula': 2000168933,
 'dia_registro': 25,
 'mes_registro': 10,
 'grupo': '2E'}

In [ ]:
estudiante['matricula']

2000168933

In [ ]:

estudiante['grupo']

'2E'

Es posible sustituir los valores dentro de una llave. Por ejemplo, recibimos la información de que el grupo del estudiante que registramos cambio a '2G' y ahora necesitamos cambiar el valor de la llave 'grupo'

In [ ]:
estudiante['grupo'] = '2G'

In [ ]:
estudiante

{'matricula': 2000168933,
 'dia_registro': 25,
 'mes_registro': 10,
 'grupo': '2G'}

También podemos añadir otros datos al diccionario. Vamos a añadir la información sobre la modalidad de estudio, nuestro estudiante estudiará inicialemente en la modalidad EAD.

Definiremos una llave llamada 'modalidad' con el valor 'EAD'.

In [ ]:
estudiante['modalidad'] = 'EAD'

In [ ]:

estudiante

{'matricula': 2000168933,
 'dia_registro': 25,
 'mes_registro': 10,
 'grupo': '2G',
 'modalidad': 'EAD'}

In [ ]:

len(estudiante)

5

# Profundizando en diccionarios
# pop()
Remueve un item de un diccionario y lo representa en la salida.

In [ ]:
estudiante.pop('grupo')

'2G'

In [ ]:
estudiante

{'matricula': 2000168933,
 'dia_registro': 25,
 'mes_registro': 10,
 'modalidad': 'EAD'}

In [ ]:
len(estudiante)

4

# items()
Retorna una lista de pares llave-valor del diccionario.

In [ ]:
estudiante.items()


dict_items([('matricula', 2000168933), ('dia_registro', 25), ('mes_registro', 10), ('modalidad', 'EAD')])

# keys()
Retorna una lista de las llaves del diccionario.

In [ ]:
estudiante.keys()

dict_keys(['matricula', 'dia_registro', 'mes_registro', 'modalidad'])

# values()
Retorna una lista de los valores del diccionario.

In [ ]:
estudiante.values()

dict_values([2000168933, 25, 10, 'EAD'])

# Lectura de valores con el lazo for

In [ ]:
for llave in estudiante.keys():
  print(llave)

matricula
dia_registro
mes_registro
modalidad


In [ ]:
for llave,valor in estudiante.items():
  print(llave,' --> ', valor)

matricula  -->  2000168933
dia_registro  -->  25
mes_registro  -->  10
modalidad  -->  EAD


# VAMOS POL!!!